oda_api/api.py

```diff
@@ -385,13 +404,13 @@ def parameters_dict_payload(self):
         }
 
         if self.is_submitted:
-            return {
-                **p,
-                'job_id': self.job_id,
-                'query_status': self.query_status,
-            }
+             return {
+                 **p,
+        #         'job_id': self.job_id,
```

Comment:
An alternative could be to re-assign the `job_id` with the one returned from the dispatcher call, if any.I notived how the one returned from the dispatcher is not actually used

Reply:
Perhaps here

[https://github.com/oda-hub/oda_api/blob/a124d8fd8e96858962a34495598df909e6c6cc5b/oda_api/api.py#L291-L342](https://github.com/oda-hub/oda_api/blob/a124d8fd8e96858962a34495598df909e6c6cc5b/oda_api/api.py#L291-L342)

At the completion of the request

Reply:
I think applying the change here would be ideal:

[https://github.com/oda-hub/oda_api/blob/a124d8fd8e96858962a34495598df909e6c6cc5b/oda_api/api.py#L507-L508](https://github.com/oda-hub/oda_api/blob/a124d8fd8e96858962a34495598df909e6c6cc5b/oda_api/api.py#L507-L508)

Reply:
How will reassigning the job_id after the request prevent sending incorrect, previously used, job_id in the next call?

Reply:
I will however add a check on returned job_id, in case it is different from the currently recorded.
If we want to keep sending job_id sometimes, we have to reset it when new parameters are set.  But it's difficult to know for sure when they changed, they might have been replaced with identical ones.

Reply:
Because that will be resent with the following request.

> I will however add a check on returned job_id, in case it is different from the currently recorded.
Exactly, that was indeed the check that I had in mind.
> If we want to keep sending job_id sometimes, we have to reset it when new parameters are set. But it's difficult to know for sure when they changed, they might have been replaced with identical ones.
> 
> http://www.faqs.org/rfcs/rfc1738.html

That is true, but the `job_id` is generated without considering the token as well as other parameters. And, in case a token is provided, the username is used.

Reply:
> Because that will be resent with the following request.

But it will be reset already after the faulty request.

> 
> > I will however add a check on returned job_id, in case it is different from the currently recorded.
> 
> Exactly, that was indeed the check that I had in mind.

This just a safety check, it will either do nothing or fail.

> 
> > If we want to keep sending job_id sometimes, we have to reset it when new parameters are set. But it's difficult to know for sure when they changed, they might have been replaced with identical ones.
> 
> That is true, but the `job_id` is generated without considering the token as well as other parameters. And, in case a token is provided, the username is used.

indeed, this makes it even worse. oda_api can not reasonably construct job_id except by calling dispatcher. Which is why it should not assume to derive it.